app/src/main/java/com/chouten/app/data/WebviewHandler.kt

```diff
@@ -2,358 +2,291 @@ package com.chouten.app.data
+            });
+        }
+
+        function sendResult(result, last = false) {
+            const currentReqId = (++reqId).toString();
+
+            // @ts-ignore
+            Native.sendHTTPRequest(JSON.stringify({
+                reqId: currentReqId,
+                action: "result",
+                shouldExit: last,
+                result: JSON.stringify(result)
+            }));
+        }
+
+        function sendSignal(signal, message = ""){
+            const currentReqId = (++reqId).toString();
+
+            // @ts-ignore
+            Native.sendHTTPRequest(JSON.stringify({
+                reqId: currentReqId,
+                action: signal === 0 ? "exit" : "error",
+                result: message
+            }));
+        }
+
+        function loadScript(url){
+            return new Promise((resolve, reject) => {
+                const script = document.createElement('script');
+                
+                script.src = url;
+                script.onload = resolve;
+                script.onerror = reject;
+        
+                document.head.appendChild(script);
+            });
+        }
+
+        """;
+        
+    private lateinit var callback: (String) -> Unit;
+    
+    /**
+     * We don't want the webview to close on error when it's the search page
+     */
+    private var closeOnError = true;
+
     private lateinit var webview: WebView
 
     /**
-     * Mutex for the webview.
-     * This is used to prevent multiple pieces of async code
-     * from trying to use the webview at the same time.
+     * Mutex for the webview. This is used to prevent multiple pieces of async code from trying to
+     * use the webview at the same time.
      */
     private var injectionLock: Mutex = Mutex(false)
 
     /**
-     * Mutex for the current call.
-     * This is used to prevent a piece of JS
-     * from being injected multiple times.
+     * Mutex for the current call. This is used to prevent a piece of JS from being injected
+     * multiple times.
      */
     private var callLock: Mutex = Mutex(false)
 
     /**
-     * The nextUrl is used to store the url which the webview
-     * will navigate to after the current request has been completed.
-     * It's used as a fallback and can be changed
-     * from within the JS
+     * The nextUrl is used to store the url which the webview will navigate to after the current
+     * request has been completed. It's used as a fallback and can be changed from within the JS
      */
     private var nextUrl: String = ""
 
-    /**
-     * Initialize the webview handler.
-     * This should be called before any other webview methods.
-     */
+    /** Initialize the webview handler. This should be called before any other webview methods. */
     @SuppressLint("SetJavaScriptEnabled", "JavascriptInterface")
     fun initialize(context: Context) {
         if (this::webview.isInitialized) return
+
+        WebView.setWebContentsDebuggingEnabled(true)
```

Comment:
I'm being picky; this could be conditional - only being enabled for debug builds.

Reply:
I agree. Although, I don't see anything particularly wrong with this, since it's not handling any sensitive data. I'll enable this when the dev mode is on for now.